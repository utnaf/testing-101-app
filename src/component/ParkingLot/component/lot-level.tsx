import React, { useEffect, useState } from 'react';
import { Box, CircularProgress, Typography } from '@material-ui/core';

import getClasses from './style';

export interface ILotLevel {
  total: number;
  count: number;
}

const EMERGENCY_LEVEL = 80;

export const LotLevel = (props: ILotLevel) => {
  const { total, count } = props;
  const classes = getClasses();

  const progress = (100 / total) * count;

  const handleClasses = () => {
    const processedClasses = [classes.lotLevel];

    if (progress >= EMERGENCY_LEVEL) {
      processedClasses.push(classes.lotLevelEmergency);
    }

    return processedClasses.join(' ');
  };

  return (
    <Box id="lot-level-box" display="flex" justifyContent="center" flexDirection="column" className={handleClasses()}>
      <CircularProgress className={classes.circularProgress} variant="static" value={progress} />
      <Box mt={1}>
        <Typography align="center">
          {count} / {total}
        </Typography>
      </Box>
    </Box>
  );
};

export default ILotLevel;
