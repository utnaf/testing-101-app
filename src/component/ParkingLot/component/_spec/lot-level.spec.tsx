import React from 'react';
import { shallow } from 'enzyme';
import { LotLevel } from '../lot-level';

const baseProps = {
  total: 100,
  count: 10,
};

let mountedWrapper = undefined;
const wrapper = (props = baseProps) => {
  if (!mountedWrapper) {
    mountedWrapper = shallow(<LotLevel {...props} />);
  }
  return mountedWrapper;
};

beforeEach(() => {
  mountedWrapper = undefined;
});

describe('<LotLevel />', () => {
  it('Renders the component to match the snapshot', () => {
    const component = wrapper();

    expect(component).toMatchSnapshot();
  });

  it('Become red when count is more than EMERGENCY_LEVEL', () => {
    const component = wrapper({
      ...baseProps,
      count: 90,
    });

    expect(component.find('#lot-level-box').hasClass(/lotLevelEmergency/)).toBe(true);
  });

  it('Become red when count is equals to EMERGENCY_LEVEL', () => {
    const component = wrapper({
      ...baseProps,
      count: 80,
    });

    expect(component.find('#lot-level-box').hasClass(/lotLevelEmergency/)).toBe(true);
  });

  it('Become red when count is equals to EMERGENCY_LEVEL', () => {
    const component = wrapper({
      ...baseProps,
      count: 10,
    });

    expect(component.find('#lot-level-box').hasClass(/lotLevelEmergency/)).toBe(false);
  });
});
