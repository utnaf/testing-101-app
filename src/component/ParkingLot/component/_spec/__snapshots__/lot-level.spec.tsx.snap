// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LotLevel /> Renders the component to match the snapshot 1`] = `
<Styled(MuiBox)
  className="makeStyles-lotLevel-2"
  display="flex"
  flexDirection="column"
  id="lot-level-box"
  justifyContent="center"
>
  <WithStyles(ForwardRef(CircularProgress))
    className="makeStyles-circularProgress-4"
    value={10}
    variant="static"
  />
  <Styled(MuiBox)
    mt={1}
  >
    <WithStyles(ForwardRef(Typography))
      align="center"
    >
      10
       / 
      100
    </WithStyles(ForwardRef(Typography))>
  </Styled(MuiBox)>
</Styled(MuiBox)>
`;
